va/http.go

```diff
@@ -664,7 +664,7 @@ func (va *ValidationAuthorityImpl) validateHTTP01(ctx context.Context, ident ide
 	payload := strings.TrimRightFunc(string(body), unicode.IsSpace)
 
 	if payload != challenge.ProvidedKeyAuthorization {
-		problem := probs.Unauthorized(fmt.Sprintf("The key authorization file from the server did not match this challenge %q != %q",
+		problem := probs.Unauthorized(fmt.Sprintf("The key authorization file from the server did not match this challenge: %q (expected %q)",
```

Comment:
I think you would need to swap `challenge.ProvidedKeyAuthorization` and `payload` for this to convey the right message.

How about the following:

```suggestion
		problem := probs.Unauthorized(fmt.Sprintf("The key authorization file from the server did not match this challenge. Expected %q (got %q)",
```